autoload/kite/codenav.vim

```diff
@@ -0,0 +1,38 @@
+function! kite#codenav#from_file()
+  call kite#codenav#request_related(expand('%:p'), v:null)
+endfunction
+
+
+function! kite#codenav#from_line()
+  if getline(".") ==# ''
+    echohl WarningMsg | echo "Code finder only works on non-empty lines."
+    return
+  endif
+  call kite#codenav#request_related(expand('%:p'), line("."))
+endfunction
+
+
+function! kite#codenav#request_related(filename, line)
+  let json = json_encode({
+    \ 'editor': 'vim',
+    \ 'location': {'filename': a:filename, 'line': a:line}
+    \ })
+  call kite#client#request_related(json, function('kite#codenav#handler'))
+endfunction
+
+
+function! kite#codenav#handler(response) abort
+  if a:response.status != 200
+    if a:response.status == 0
+      echohl WarningMsg | echo "Kite could not be reached. Please check that Kite Engine is running."
```

Comment:
I'm not sure this is semantically what's going on, but it does capture the unreachable case.

Reply:
Yes, `0` response code means we couldn't get a usable response from Kite.